specs/src/tx_client_v2.md

```diff
@@ -0,0 +1,207 @@
+# Tx Client v2 – Sequential Submission Queue
+
+## Abstract
+
+The biggest pain point and bottleneck preventing 100% successful submissions is a race condition caused by eviction handling in tandem with our recovery mechanism in broadcast:
+
+1. A transaction **Tx(3)** is **evicted** from the mempool.
+2. The **Tx client does not poll** quickly enough to detect the eviction.
```

Comment:
should we just increase the polling frequency to catch this? or check if the previous transaction was evicted before sending the new one?

Reply:
We could increase polling frequency but if we hit sequence mismatch when we're trying to broadcast we'd also check if previous tx was evicted and we're waiting for it to be resubmitted.